lib/template-generator.js

```diff
@@ -1,51 +1,51 @@
 'use strict'
-var path = require('path')
-var fs = require('fs')
-var slug = require('slug')
-var formatDate = require('dateformat')
-var mkdirp = require('mkdirp')
 
-module.exports = function templateGenerator (opts, cb) {
+const path = require('path')
+const fs = require('fs')
+const util = require('util')
+const slug = require('slug')
+const formatDate = require('dateformat')
+const mkdirp = require('mkdirp')
+
+const pReadFile = util.promisify(fs.readFile)
+const pWriteFile = util.promisify(fs.writeFile)
+const pMkdirp = util.promisify(mkdirp)
```

Comment:
For some reason, with the newest versions of the dependencies, this was throwing an error. 

```
Template generation error : invalid options argument
```

I replaced the `util.promisify` call with the following:

```js
const pMkdirp = async (dir) => {
  const promise = new Promise((resolve, reject) => {
    return mkdirp(dir)
      .then(res => resolve(res))
      .catch(err => reject(err))
  })
  return await promise
}
```

And got it working. FYI.

Reply:
Alternatively you could probably just use https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fspromises_readfile_path_options

You are more than welcome to submit a PR to https://github.com/jdpnielsen/node-migrate

Further discussion should probably also be moved there.

Reply:
Thanks for the suggestion, I'll give it a shot. Started watching node-migrate as well and will help where I can!